aten/src/ATen/native/native_functions.yaml

```diff
@@ -2062,6 +2062,12 @@
     CPU: _fft_r2c_mkl
     CUDA: _fft_r2c_cufft
 
+- func: _fft_r2c.out(Tensor self, int[] dim, int normalization, bool onesided, *, Tensor(a!) out) -> Tensor(a!)
```

Comment:
I imagine this will run afoul of https://github.com/pytorch/pytorch/pull/49510 and need an update.

cc @smessmer

Reply:
When I wrote this I ha to drop `use_c10_dispatcher: full` because the out argument can't go at the end since it doesn't have a default. Looks like `hacky_wrapper_for_legacy_signatures` works fine though and it compiles after rebasing on #49510.